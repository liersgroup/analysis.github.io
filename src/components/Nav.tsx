import React from 'react'

import { Container, Nav, Navbar } from 'react-bootstrap'
import { Link } from 'react-router-dom'
import paths from '../paths'
import { MAIN_BRANCH } from '../models/Store'
import { FaCodeBranch } from 'react-icons/fa'


const DevLink = ({ branch }: { branch: string }) => {
  const contents = branch === MAIN_BRANCH
    ? 'Dev'
    : (
      <>
        <FaCodeBranch />
        {' '}
        {branch}
      </>
    )

  return (
    <Link
      className="nav-link"
      to="/dev"
    >
      {contents}
    </Link>
  )
}

export default React.memo(
  function Navigation({ branch }: { branch: string }) {
    const showDevLink = process.env.NODE_ENV === 'development'
      || window.location.host.includes('dev.') || window.location.host.includes('development.')
      || branch !== MAIN_BRANCH

    return (
      <Navbar bg={branch === MAIN_BRANCH ? "light" : "dark"}>
        <Container>
          <Navbar.Brand>
            <Link to="/">Ï€-Base</Link>
          </Navbar.Brand>
          <Navbar.Toggle aria-controls="navbar-nav" />
          <Navbar.Collapse id="navbar-nav">
            <Nav>
              <Link className="nav-link" to="/spaces">Spaces</Link>
              <Link className="nav-link" to="/properties">Properties</Link>
              <Link className="nav-link" to="/theorems">Theorems</Link>
            </Nav>
            <Nav className="ml-auto">
              {showDevLink && <DevLink branch={branch} />}
              <a className="nav-link" href={paths.contributingGuide()}>Contribute</a>
            </Nav>
          </Navbar.Collapse>
        </Container>
      </Navbar>
    )
  }
)
