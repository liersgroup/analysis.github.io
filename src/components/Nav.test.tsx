import React from 'react'
import { MemoryRouter as Router } from 'react-router-dom'
import { render } from 'enzyme'

import Nav from './Nav'

describe('Nav', () => {
  it('renders correctly', () => {
    const component = render(
      <Router>
        <Nav />
      </Router>
    )

    expect(component).toMatchSnapshot()
  })
})
