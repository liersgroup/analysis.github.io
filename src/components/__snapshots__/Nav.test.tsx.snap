// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders correctly 1`] = `
<nav
  class="navbar navbar-expand navbar-light bg-dark"
>
  <div
    class="container"
  >
    <span
      class="navbar-brand"
    >
      <a
        href="/"
      >
        Ï€-Base
      </a>
    </span>
    <button
      aria-controls="navbar-nav"
      aria-label="Toggle navigation"
      class="navbar-toggler collapsed"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="navbar-collapse collapse"
      id="navbar-nav"
    >
      <div
        class="navbar-nav"
      >
        <a
          class="nav-link"
          href="/spaces"
        >
          Spaces
        </a>
        <a
          class="nav-link"
          href="/properties"
        >
          Properties
        </a>
        <a
          class="nav-link"
          href="/theorems"
        >
          Theorems
        </a>
      </div>
      <div
        class="ml-auto navbar-nav"
      >
        <a
          class="nav-link"
          href="/dev"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 384 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M384 144c0-44.2-35.8-80-80-80s-80 35.8-80 80c0 36.4 24.3 67.1 57.5 76.8-.6 16.1-4.2 28.5-11 36.9-15.4 19.2-49.3 22.4-85.2 25.7-28.2 2.6-57.4 5.4-81.3 16.9v-144c32.5-10.2 56-40.5 56-76.3 0-44.2-35.8-80-80-80S0 35.8 0 80c0 35.8 23.5 66.1 56 76.3v199.3C23.5 365.9 0 396.2 0 432c0 44.2 35.8 80 80 80s80-35.8 80-80c0-34-21.2-63.1-51.2-74.6 3.1-5.2 7.8-9.8 14.9-13.4 16.2-8.2 40.4-10.4 66.1-12.8 42.2-3.9 90-8.4 118.2-43.4 14-17.4 21.1-39.8 21.6-67.9 31.6-10.8 54.4-40.7 54.4-75.9zM80 64c8.8 0 16 7.2 16 16s-7.2 16-16 16-16-7.2-16-16 7.2-16 16-16zm0 384c-8.8 0-16-7.2-16-16s7.2-16 16-16 16 7.2 16 16-7.2 16-16 16zm224-320c8.8 0 16 7.2 16 16s-7.2 16-16 16-16-7.2-16-16 7.2-16 16-16z"
            />
          </svg>
           
        </a>
        <a
          class="nav-link"
          href="https://github.com/pi-base/data/blob/master/CONTRIBUTING.md"
        >
          Contribute
        </a>
      </div>
    </div>
  </div>
</nav>
`;
